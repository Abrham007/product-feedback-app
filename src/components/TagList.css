.TagList {
  width: 255px;
  height: 166px;
  display: flex;
  gap: 8px;
  flex-wrap: wrap;
  padding: 24px;
  border-radius: 10px;
  background: #fff;
}

@media screen and (max-width: 1120px) {
  .TagList {
    width: 223px;
    height: 178px;
  }
}
