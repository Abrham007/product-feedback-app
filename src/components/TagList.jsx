import "./TagList.css";
import Tag from "./Tag";
import { useState } from "react";
import { useSelector } from "react-redux";
import { selectSuggestionList } from "../features/productRequests/productRequestsSlice";

function TagList(props) {
  const [selectedTag, setSelectedTag] = useState("all");
  const tagList = ["All", "UI", "UX", "Enhancement", "Bug", "Feature"];
  const suggestions = useSelector(selectSuggestionList);

  function onTagClick(category) {
    if (category === "all") {
      props.handleSuggestions(
        suggestions.filter((req) => req.status === "suggestion")
      );
    } else {
      props.handleSuggestions(
        suggestions.filter(
          (req) => req.status === "suggestion" && req.category === category
        )
      );
    }
    setSelectedTag(category);
  }
  return (
    <div className="TagList">
      {tagList.map((item) => (
        <Tag
          key={item}
          text={item}
          isSelected={selectedTag}
          onClick={onTagClick}
        />
      ))}
    </div>
  );
}

export default TagList;
