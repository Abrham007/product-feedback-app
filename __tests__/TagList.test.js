import renderer from "react-test-renderer";
import TagList from "../src/components/TagList";

describe("TagList", () => {
  it("renders the correct JSX when first mounted ", () => {
    const component = renderer.create(
      <TagList handleSuggestions={() => {}} suggestions={[]} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders the correct JSX after user clickes tag", () => {
    const component = renderer.create(
      <TagList handleSuggestions={() => {}} suggestions={[]} />
    );
    let root = component.root;
    renderer.act(() => {
      root.findAllByType("button")[3].props.onClick();
    });
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
