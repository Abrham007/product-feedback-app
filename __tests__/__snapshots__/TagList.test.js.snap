// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList renders the correct JSX after user clickes tag 1`] = `
<menu
  className="TagList"
>
  <button
    className="Tag"
    disabled={false}
    onClick={[Function]}
    style={{}}
  >
    All
  </button>
  <button
    className="Tag"
    disabled={false}
    onClick={[Function]}
    style={{}}
  >
    UI
  </button>
  <button
    className="Tag"
    disabled={false}
    onClick={[Function]}
    style={{}}
  >
    UX
  </button>
  <button
    className="Tag"
    disabled={false}
    onClick={[Function]}
    style={
      {
        "backgroundColor": "#4661e6",
        "color": "#fff",
      }
    }
  >
    Enhancement
  </button>
  <button
    className="Tag"
    disabled={false}
    onClick={[Function]}
    style={{}}
  >
    Bug
  </button>
  <button
    className="Tag"
    disabled={false}
    onClick={[Function]}
    style={{}}
  >
    Feature
  </button>
</menu>
`;

exports[`TagList renders the correct JSX when first mounted  1`] = `
<menu
  className="TagList"
>
  <button
    className="Tag"
    disabled={false}
    onClick={[Function]}
    style={
      {
        "backgroundColor": "#4661e6",
        "color": "#fff",
      }
    }
  >
    All
  </button>
  <button
    className="Tag"
    disabled={false}
    onClick={[Function]}
    style={{}}
  >
    UI
  </button>
  <button
    className="Tag"
    disabled={false}
    onClick={[Function]}
    style={{}}
  >
    UX
  </button>
  <button
    className="Tag"
    disabled={false}
    onClick={[Function]}
    style={{}}
  >
    Enhancement
  </button>
  <button
    className="Tag"
    disabled={false}
    onClick={[Function]}
    style={{}}
  >
    Bug
  </button>
  <button
    className="Tag"
    disabled={false}
    onClick={[Function]}
    style={{}}
  >
    Feature
  </button>
</menu>
`;
